### DISCUSSION TITLE
Add option for custom network name prefix

### DISCUSSION THREAD

**Discussion by oktalz** [testcontainers/testcontainers-go#2123] [Ideas]
Hi.
new API for network does not allow custom name for network, 
if running multiple tests, its hard to differentiate what test is using what network.
especially if more that one project is running tests.
this is also important for stats collection... etc.

Also, ryuk have some issues with deleting networks (in many occasions),
so while its easy to see with labels what is what, 
manually is more work.

While this is irrelevant in most cases and we have labels, its much nicer to have distinction between projects/tests.

Would adding a `WithPrefix` or and `WithName` to the api cause some issue and does it seems interesting ?
like `network.WithPrefix('proj_name')`

**Comment by mdelapenya** [comment]
Hi @oktalz sorry for coming back to this late. Yeah, I like your proposal. Given the options are that, optional, users will be free to choose fully randomised names or having prefixes. But I'd enforce not having fixed network names but as you proposed: only prefixes. If we add a `WithName` although optional, could cause inconsistencies in the experience of using it.

**Reply by oktalz** [reply]
OK, I can start with WithPrefix, reason I wanted to use WithName too is because now, for name `uuid` is used (I kind of like ULID since its shorter and sortable), so it limits options - there is a limit on network name length too.

anyways, I'll create a PR to add WithPrefix, WithName can be left to be done later (if even)

I was also considering adding WithULID (that would replace UUID), for example [oklog/ulid](https://github.com/oklog/ulid), but that would add another dependancy so its a bit more to add (Licence is compatible though). IMHO that is a bit to much, and WithName would leave that to the end user

**Reply by mdelapenya** [reply]
Sounds good to me.

Regarding the replacement of google's UUID package for oklog/ulid, I'd let it for separate discussion, as a follow-up.

**Comment by woltere** [comment]
I would like to configure a prefix because when multiple networks are configured the gateway is selected on lexicographically ordering of the network name.
